### ISSUE TITLE
plan_conv

### ISSUE THREAD

**Issue by baggepinnen** [JuliaDSP/DSP.jl#358]
When performing several similar convolutions, it would be convenient to have a facility similar to `plan_fft` that creates a workspace structure with caches for `conv`, so that `conv!(plan, u, v)` or `conv!(out, plan, u, v)` is allocation free

**Comment by galenlynch** [comment]
Yeah that could be useful, and maybe not all that difficult to do.

**Comment by JanJereczek** [comment]
>  Might be good for us to consider how to separate that from a standalone plan if / when we implement some form of `plan_conv`

Sure, this makes a lot of sense for other applications.

> or even use GPUs for the whole thing

I actually do with minor adaptations of the code above!

> Do you mind sharing what you're doing with this?

I am computing the elastic deformation of the solid Earth, which can be computed as a convolution of a Green's function with the surface load (essentially oceans and ice sheets). The load changes (in a pretty large time loop) but the Green's function remains the same of course. The code above is tailored for this but note that it is a quite common thing in physics that your kernel/Green's function remains the same and that you repeatedly convolve it with a time-evolving boundary condition. As far as I understand, this goes beyond physics since it applies to any response of a time-invariant linear system, so the applicability of a precomputed plan + kernel could be larger. Maybe it would make sense to multiple dispatch `plan_conv(...)` and `plan_conv(..., kernel)` so that it distinguishes between a potentially changing or a fixed kernel. What do you think?

**Comment by JanJereczek** [comment]
Sure! Thanks for asking though ;)
